File: src/python/_1220_Count_Vowels_Permutation.py
class Solution:
    def countVowelPermutation(self, n: int) -> int:
        MOD = 10**9 + 7

        a, e, i, o, u = 1, 1, 1, 1, 1

        for _ in range(n - 1):
            A = e
            E = (a + i) % MOD
            I = (a + e + o + u) % MOD
            O = (i + u) % MOD
            U = a

            a, e, i, o, u = A, E, I, O, U

        return (a + e + i + o + u) % MOD

File: src/java/_1220_Count_Vowels_Permutation.java
class Solution {
    public int countVowelPermutation(int n) {
        final int MOD = 1000000007;

        long a = 1, e = 1, i = 1, o = 1, u = 1;

        for (int idx = 0; idx < n - 1; idx++) {
            long A = e;
            long E = (a + i) % MOD;
            long I = (a + e + o + u) % MOD;
            long O = (i + u) % MOD;
            long U = a;

            a = A;
            e = E;
            i = I;
            o = O;
            u = U;
        }

        return (int) ((a + e + i + o + u) % MOD);
    }
}
